src/fractionatedmorse.rs

```diff
@@ -0,0 +1,333 @@
+    /// if an unsupported symbol is present. The support characters are `a-z`, `A-Z`, `0-9`, spaces
+    /// and the special characters `@ ( ) . , : ' " ! ? - ; =`.
+    fn encrypt_morse(message: String) -> Result<String, &'static str> {
+        let mut morse = String::new();
+
+        // Convert each letter in message to corresponding Morse characters.
+        // We cannot have multiple spaces in a row, otherwise the invalid Fractionated Morse
+        // character `|||` will be produced. Therefore, split by whitespace and only add one space
+        // in-between each word.
+        for word in message.split_whitespace() {
+            for c in word.chars() {
+                if let Some(pos) = MORSE_ALPHABET.iter().position(|m| m.0 == c.to_ascii_lowercase()) {
+                    morse.extend(MORSE_ALPHABET[pos].1.chars());
+                    morse.push('|');
+                } else {
+                    return Err("Invalid message. Please strip any unsupported symbols.")
+                }
+            }
+            // Add a single space between each word. Spaces are represented having nothing
+            // between two seperators, `||`, so we push another seperator here. This also takes
+            // care of the final `||` required to end the message.
+            morse.push('|');
+        }
+
+        Ok(morse)
+    }
+
+    /// Takes a Morse code string, with each Morse character separated by `|`, and converts it to
+    /// plaintext. This function returns `Err` if an invalid Morse character is encountered.
+    fn decrypt_morse(mut message: String) -> Result<String, &'static str> {
+        let mut plaintext = String::new();
+
+        // Remove character separators from the beginning of the message if present
+        while message.starts_with('|') {
+            message.remove(0);
+        }
+
+        // Loop over every Morse character
+        for morse_chr in message.split('|') {
+            // Find the Morse character in the alphabet and decode it.
+            if let Some(pos) = MORSE_ALPHABET.iter().position(|&m| m.1 == morse_chr) {
+                plaintext.push(MORSE_ALPHABET[pos].0);
+            } else {
+                return Err("Invalid Fractionated Morse message. Unknown Morse character found.")
+            }
+        }
+
+        // Messages are ended with `||` but this will be decoded as a two spaces in the loop above.
+        // Therefore remove any trailing whitespace. This will also remove any whitespace present
+        // at the end of the plaintext.
+        plaintext = plaintext.trim_right().to_string();
+
+        Ok(plaintext)
+    }
+
+    /// Takes a alphabetic string and converts it to Fractionated Morse. This function will
+    /// return `Err` if a non-alphabetic symbol is present in the message.
+    fn encrypt_frac_morse(keyed_alphabet: &String, message: String) -> Result<String, &'static str> {

```

Comment:
perhaps change this to `to_fractionated_morse` to make it clear that we are going from text -> fractionated_morse